addon/mixins/axis-titles.js

```diff
@@ -0,0 +1,174 @@
+    if(this.get('hasAxisTitles')){
+      return this.get('axisTitleHeight') + this.get('labelPadding');
+    }else{
+      return 0;
+    }
+  }),
+
+  /**
+   * References and/or creates the d3 element for x axis title
+   * @type {Object}
+   */
+  xAxisTitle: Ember.computed(function() {
+    return this.selectOrCreateAxisTitle('.x.axis-title').attr('class', 'x axis-title');
+  }).volatile(),
+
+  /**
+   * References and/or creates the d3 element for y axis title
+   * @type {Object}
+   */
+  yAxisTitle: Ember.computed(function() {
+    return this.selectOrCreateAxisTitle('.y.axis-title').attr('class', 'y axis-title');
+  }).volatile(),
+
+  /**
+   * Amount of padding for the axis, `labelHeightOffset` + `labelPadding`
+   * @type {Number}
+   */
+  axisPadding: Ember.computed('labelHeightOffset', 'labelPadding', function(){
+    return this.get('labelHeightOffset') + this.get('labelPadding');
+  }),
+
+  /**
+   * Position of x axis title on the x axis
+   * @type {Number}
+   */
+  xAxisPositionX: Ember.computed('graphicWidth', 'labelWidthOffset', function(){
+    return this.get('graphicWidth') / 2 + this.get('labelWidthOffset');
+  }),
+
+  /**
+   * Position of x axis title on the y axis
+   * @type {Number}
+   */
+  xAxisPositionY: Ember.computed('graphicBottom', 'axisPadding', function(){
+    return this.get('graphicBottom') + this.get('axisPadding');
+  }),
+
+  /**
+   * Position of y axis title on the x axis
+   * @type {Number}
+   */
+  yAxisPositionX: Ember.computed('graphicHeight', 'labelWidthOffset', function(){
+    return -(this.get('graphicHeight') / 2 + this.get('labelWidthOffset'));
+  }),
+
+  /**
+   * Position of y axis title on the y axis
+   * @type {Number}
+   */
+  yAxisPositionY: -20,
+
+  /**
+   * X axis transform
+   * @type {string}
+   */
+  xAxisTransform: "rotate(0)",
+  /**
+   * Y axis transform
+   * @type {string}
+   */
+  yAxisTransform: "rotate(-90)",
+
+  /**
+   * If no axis title has been created for the selector create a new one
+   * @param  {String} selector
+   * @return {Object}
+   */
+  selectOrCreateAxisTitle: function(selector) {
+    var title = this.get('viewport').select(selector);
+    if (title.empty()) {
+      return this.get('viewport').append('text');
+    } else {
+      return title;
+    }
+  },
+
+  /**
+   * Update the x axis title and position
+   */
+  updateXAxisTitle: function(){
+    this.get('xAxisTitle').text(this.get('xAxisTitleDisplayValue')).style('text-anchor', 'middle').attr({
+      x: this.get('xAxisPositionX'),
+      y: this.get('xAxisPositionY')
+    });
+  },
+
+  /**
+   * Update the y axis title and position
+   */
+  updateYAxisTitle: function(){
+    this.get('yAxisTitle').text(this.get('yAxisTitleDisplayValue')).style('text-anchor', 'start').attr({
+      x: this.get('yAxisPositionX'),
+      y: this.get('yAxisPositionY'),
+    }).attr("transform", this.get('yAxisTransform')).attr("dy", "1em");
+  },
```

Comment:
`updateXAxisTitle` and `updateYAxisTitle` seem like they could share a common helper function. Not sure if it'd make things nicer or not though.